gtsam/linear/GaussianBayesNet.cpp

```diff
@@ -64,8 +64,9 @@ namespace gtsam {
     return sample(result, rng);
   }
 
-  VectorValues GaussianBayesNet::sample(VectorValues result,
+  VectorValues GaussianBayesNet::sample(const VectorValues& given,
```

Comment:
Ok, for this case, the question is whether it should be value (which will automatically make the copy you inserted below) or by const&, which you modified this code to. I chose by value when implementing this so I would get the automatic copy, but maybe it does not make a difference and it is more “clever than smart”.  The code that was here was correct, though.

Reply:
I agree it was correct. I felt this made it clearer to read that you're `given` some VectorValues and you're getting back the `result` of sampling.
I can undo this if you'd like.